src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java

```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE deleted = FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            List<Manufacturer> manufacturers = new ArrayList<>();
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Manufacturer manufacturer = new Manufacturer(name, country);
+                manufacturer.setId(id);
+                manufacturers.add(manufacturer);
+            }
+            statement.close();
+
+            return manufacturers;
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Couldn't establish connection to MySQL server. "
+                    + throwables);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?, country = ?"
+                + " WHERE id = ? AND deleted = FALSE";
+        String getNewQuery = "SELECT * FROM manufacturers WHERE id = " + manufacturer.getId();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement
+                    = setUpdate(connection.prepareStatement(query), manufacturer);
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(getNewQuery);
+            Manufacturer manufacturerNew = null;
+            if (resultSet.next()) {
+                manufacturerNew
+                        = new Manufacturer(resultSet.getString("name"),
+                                resultSet.getString("country"));
+                manufacturerNew.setId(resultSet.getObject("id", Long.class));
+            }
+            if (manufacturerNew == null) {
+                throw new RuntimeException("Couldn't get new manufacturer");
+            }
+            statement.close();
+
+            return manufacturerNew;
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Couldn't establish connection to MySQL server. "
+                    + throwables);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturers SET deleted = TRUE WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
```

Comment:
Create constants for magic

Reply:
I can't, cause this indexes don't serve the same purpose every time they are used.